### ISSUE TITLE
@InputArgument to default to parameter name

### ISSUE THREAD

**Issue by aramir** [Netflix/dgs-framework#93]
It would be great for `@InputArgument`'s `value` to default parameter name -- similar to Spring Framework's `@PathVariable` and `@RequestParam` annotations. So we can compact
```java
public List<Show> shows(@InputArgument("titleFilter") String titleFilter)
```
to
```java
public List<Show> shows(@InputArgument String titleFilter)
```

**Comment by paulbakker** [comment]
@aramir I thought parameter name reflection is only possible if the `-parameters` compiler flag is set. I'm now reading that with Spring Boot it _does_ work out of the box. Is that correct?

In that case I'm definitely in favor of support this.

**Comment by aramir** [comment]
It actually works even with vanilla Spring, without Spring Boot. And yes - out of the box, no flag needed.

**Comment by aramir** [comment]
After looking into it, Spring uses an implementation of [ParameterNameDiscoverer](https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/core/ParameterNameDiscoverer.html) interface called [DefaultParameterNameDiscoverer](https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/core/DefaultParameterNameDiscoverer.html) which, internally sequentially tries a few different methods to figure out parameter names. 

It *does* depend on either `-g:vars` or `-parameter`, however, both maven and gradle set those for java compile processes by default.

More info:

 - [Spring MVC - Automatic Parameter Names Discovery](https://www.logicbig.com/tutorials/spring-framework/spring-web-mvc/parameter-name-discoverer.html)
 - [How Spring gets parameter names without debug information](https://stackoverflow.com/questions/36466913/how-spring-gets-parameter-names-without-debug-information)

**Comment by paulbakker** [comment]
Fixed in #112
